### PR TITLE
Docs: Added Solutions to Leetcode 609

### LINKED ISSUE #1376: [Feature Request]: Leetcode Problem 609 - (Find Duplicate File in System-Medium)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

Provide Solution To Leetcode Problem 609 - (Find Duplicate File in System)

### Use Case

I would add proper solutions for the problem in three different programing languages i.e C++,Java,and Python with proper documentation

### Benefits

Users can access solution to problem 609

### Add ScreenShots

N/A

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by shreyash3087**
## Related Issue

Closes #1376

## Description

This PR adds the solution to leetcode problem 609 - (Find Duplicate File in System)

## Type of PR

- [ ] Bug fix
- [ ] Feature enhancement
- [X] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

screencapture-localhost-3000-dsa-solutions-lc-solutions-0600-0699-find-duplicate-file-in-system-2024-06-16-18_26_26

## Checklist
- [X] I have performed a self-review of my code.
- [X] I have read and followed the Contribution Guidelines.
- [X] I have tested the changes thoroughly before submitting this pull request.
- [X] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have followed the code style guidelines of this project.
- [X] I have checked for any existing open issues that my pull request may address.
- [X] I have ensured that my changes do not break any existing functionality.
- [X] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [X] I have read the resources for guidance listed below.
- [X] I have followed security best practices in my code changes.

## Additional Context

N/A

### CODE REVIEW

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+**Example 2:**
+
+```
+Input: paths = ["root/a 1.txt(abcd) 2.txt(efgh)","root/c 3.txt(abcd)","root/c/d 4.txt(efgh)"]
+Output: [["root/a/2.txt","root/c/d/4.txt"],["root/a/1.txt","root/c/3.txt"]]
+```
+
+### Constraints
+
+- $1 <= paths.length <= 2 * 10^4$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:46]: ### basic practice

```md
$1 <= paths.length <= 2 * 10^4$
```

### Good Practice

```md
$1 \leq \text{paths.length} \leq  2 \times 10^4$
```

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+- $1 <= paths[i].length <= 3000$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:47]: also, replace `- $1 <= paths[i].length <= 3000$` to best or Good practice

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+- $1 <= sum(paths[i].length) <= 5 * 10^5$
+- `paths[i]` consist of English letters, digits, `'/'`, `'.'`, `'('`, `')'`, and `' '`.
+- You may assume no files or directories share the same name in the same directory.
+- You may assume each given directory info represents a unique directory. A single blank space separates the directory path and file info.
+
+## Solution for Find Duplicate File in System
+
+### Approach #1 Brute Force [Time Limit Exceeded]
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:55]: replace `#1` to `1`

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+
+### Time Complexity: $O(n*x + f^2 * s)$
+
+> **Reason**: Creation of list will take O(n∗x), where n is the number of directories and x is the average string length. Every file is compared with every other file. Let f files are there with average size of s, then files comparision will take O(f2∗s), equals can take O(s). Here, Worst case will be when all files are unique.
+
+### Space Complexity: $O(n*x)$
+
+> **Reason**: Size of lists res and list can grow upto n∗x.
+
+### Approach #2 Using HashMap
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:203]: replace `#2` to `2`

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+</TabItem>
+</Tabs>
+
+## Complexity Analysis
+
+### Time Complexity: $O(n*x + f^2 * s)$
+
+> **Reason**: Creation of list will take O(n∗x), where n is the number of directories and x is the average string length. Every file is compared with every other file. Let f files are there with average size of s, then files comparision will take O(f2∗s), equals can take O(s). Here, Worst case will be when all files are unique.
+
+### Space Complexity: $O(n*x)$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:199]: replace `$O(n*x)$` to `$O(n \times x)$`

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+                res.append(l)
+        
+        return res
+```
+</TabItem>
+</Tabs>
+
+## Complexity Analysis
+
+### Time Complexity: $O(n*x + f^2 * s)$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:195]: replace `$O(n*x + f^2 * s)$` to `$O(n \times x + f^2 \times s)$`

File: dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+</TabItem>
+</Tabs>
+
+## Complexity Analysis
+
+### Time Complexity: $O(n*x)$
+
+> **Reason**: n strings of average length x is parsed.
+
+### Space Complexity: $O(n*x)$
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/0600-0699/0609-find-duplicate-file-in-system.md:316]: also, update `$O(n*x)$` to best practice

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by shreyash3087** [comment]
@Ajay-Dhangar Done

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**